### PR TITLE
retrieve and merge post parameters in ValidateIncomingData

### PR DESCRIPTION
**Pull request by Orkin**

### CODE REVIEW

File: src/Mvc/Controller/Plugin/ValidateIncomingData.php [hunk 1]
```diff
@@ -62,7 +62,10 @@ public function __invoke($inputFilterName, array $validationGroup = [], $context
             $inputFilter->setValidationGroup($validationGroup);
         }
 
-        $data = json_decode($this->controller->getRequest()->getContent(), true) ?: [];
+        $data = array_merge(
+            json_decode($this->controller->getRequest()->getContent(), true) ?: [],
```
Comment (Ocramius) [inline src/Mvc/Controller/Plugin/ValidateIncomingData.php]: Avoiding the `getRequest()` call

**Comment by Ocramius** [comment]
The request content *IS* the POST data. What are you trying to solve here, exactly?

File: src/Mvc/Controller/Plugin/ValidateIncomingData.php (hunk 1 above)
Comment (Orkin) [inline src/Mvc/Controller/Plugin/ValidateIncomingData.php]: You prefer $this->controller->request ?

**Comment by Orkin** [comment]
When we upload a file the content of the request is multipart so we don't have a content and we need to retrieve data from $_POST

**Comment by Ocramius** [comment]
@Orkin what's the request content in that case (the output of `getContent()`)?

**Comment by Orkin** [comment]
It's empty : var_dump => string(0) ""

**Comment by Orkin** [comment]
ping @Ocramius

**Comment by bakura10** [comment]
From what I understand @Ocramius , @Orkin is using traditional form/www-multipart, hence it cannot be decoded using json_deocde.

I'm not against this change. It is a bit hacky but does the job for now.

**Comment by Orkin** [comment]
@bakura10  It's only when there is file uploading with some data ;)

**Comment by bakura10** [comment]
I'm merging now. I can understand the problem that it solves and I don't see any problems that could be introduced. I don't have time nor the courage to find a better solution for now.

**Comment by Ocramius** [comment]
Opened #197 /cc @Orkin

**Pull request merged**